cloud_function_go/function.go

```diff
@@ -0,0 +1,93 @@
+// Package p contains an HTTP Cloud Function.
+package p
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+func shouldRetry(statusCode int) bool {
+	retry := true
+	switch statusCode {
+	case http.StatusBadRequest:
+		fmt.Printf("Got HTTP %d bad request, skip retry\n", statusCode)
+		retry = false
+	case http.StatusNotFound:
+		fmt.Printf("Got HTTP %d not found, skip retry\n", statusCode)
+		retry = false
+	case http.StatusUnauthorized:
+		fmt.Printf("Got HTTP %d unauthorized, skip retry\n", statusCode)
+		retry = false
+	case http.StatusForbidden:
+		fmt.Printf("Got HTTP %d forbidden, skip retry\n", statusCode)
+		retry = false
+	case http.StatusOK:
+		retry = false
+	}
+	return retry
+}
+
+func LogzioHandler(w http.ResponseWriter, r *http.Request) {
+	type Data struct {
+		Data string `json:"data"`
+	}
+
+	var d struct {
+		Message Data `json:"message"`
+	}
+	token := r.URL.Query().Get("token")
```

Comment:
please move the config and validation part to a different function
Also the validation should come immediate after the query get and not in the end